src/components/MapFacetContents.js

```diff
@@ -199,22 +229,12 @@ class MapFacetContents extends React.Component<MapFacetContentsDefaultProps, Map
       });
 
       const points = this.props.buckets.map((bucket) => {
-        const value = bucket.value;
-        let longitude = NaN;
-        let latitude = NaN;
-        if (typeof value === 'string') {
-          const valueArr = value.split(',');
-          if (valueArr.length === 2) {
-            longitude = Number.parseFloat(valueArr[0]);
-            latitude = Number.parseFloat(valueArr[1]);
-          }
-        } else {
-          longitude = value.longitude;
-          latitude = value.latitude;
-        }
-        if (Number.isNaN(longitude) || Number.isNaN(latitude)) {
+        // Return null if getCoordinatesFromBucket() returns null value
+        const coordinates = MapFacetContents.getCoordinatesFromBucket(bucket);
+        if (!coordinates) {
```

Comment:
Actually, this is essentially doing the same mapping and filtering as the boundary code, isn't it... maybe make the `getCoordinatesFromBucket()` function take the array off buckets and return an array of the non-null coordinates (probably renaming it to `getCoordinatesFromBuckets()`)... Then, here, you can just map that list to `<Maker>` components.

Reply:
This sounds like a good approach. Although, the `<Marker>` component also requires the corresponding `bucket` for the current coordinates.
In case the new function returns an array of coordinates, we will have to again get the `bucket` for the respective coordinate when we map that array.

Reply:
You're totally right. I wasn't thinking about the need for the bucket to apply the filter.